frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -44,6 +86,15 @@ export const insightLogic = kea<insightLogicType>({
             },
         ],
     }),
+    listeners: ({ actions, values }) => ({
+        startQuery: () => {
+            values.timeout && clearTimeout(values.timeout)
+            actions.setTimeout(setTimeout(() => actions.setShowTimeoutMessage(true), 15000))
```

Comment:
Hmm are we making a hard requirement now that all queries should run under 15 seconds? There are some queries that could probably cut closer to 20-30seconds

Reply:
Two new points:
- Nit: This time should probably be a const somewhere
- The error message seems to indicate that we gave up, although if I understand correctly, the query is still running and might give results in some seconds anyway? If so, would be great to add a spinner somewhere to show it's still pending

Reply:
The spinner will still be shown after 15 seconds so there is indication that it's still loading. I can reword the message if it's still going.

Agree re: const